share/p2p/peers/manager.go

```diff
@@ -298,12 +297,21 @@ func (m *Manager) subscribeHeader(ctx context.Context, headerSub libhead.Subscri
 			log.Errorw("get next header from sub", "err", err)
 			continue
 		}
-		m.validatedPool(h.DataHash.String())
+		m.validatedPool(h.DataHash.String(), h.Height())
 
 		// store first header for validation purposes
 		if m.initialHeight.CompareAndSwap(0, h.Height()) {
 			log.Debugw("stored initial height", "height", h.Height())
 		}
+
+		// update storeFrom if header height is higher than current value
+		old := m.storeFrom.Load()
+		upd := int64(h.Height()) - amountOfStoredPools
+		if old < upd {
```

Comment:
I don't think this is necessary. Subscription guarantees that new headers are always higher than previous

Reply:
As well that debug log